### ISSUE TITLE
How to get the prediction probabilities to score the confidence of the classification?

### ISSUE THREAD

**Issue by karhunenloeve** [karhunenloeve/CNN#1]
This is for sure a question which occurred to me when I started using TensorFlow. With the graph structure and the session storage it turns out to be very easy to get the desired values. Obviously the used language is python 3, as it is in the whole tutorial.
This method is for class prediction purpose.

In `cls_pred` the predicted classes will be stored. In `cls_proba` the associated probabilities.
First create two arrays with the shape of the classified image. I used integer as datatype, but you can also think of floating point numbers. Initialize a counting variable (here `i`).
_Note: Floating point number is necessary as datatype to calculate the probabilities._
```
cls_pred = np.zeros(shape=num_images, dtype=np.int64)
cls_prob = np.zeros(shape=num_images, dtype=np.float64)
i = 0
```

Loop through the image and set a new batch size (as `j`). Pass as `x`-arguments the transfered-image values from the inception model (or any other model) and the label values for the same amount of images into the placeholder `y_true`.
```
 while i < num_images:
       j = min(i + batch_size, num_images)
       feed_dict = {x: transfer_values[i:j],
                    y_true: labels[i:j]}
```

Finally pass the desired arguments into the current session and run it. For the confidence probability run simply the `argmax` over the transfered values and divide it by the sum of the max arguments. This will give you the loss. To obtain a probability subtract the loss from one.
```
cls_pred[i:j] = session.run(y_pred_cls, feed_dict=feed_dict)
proba = 1 - (tf.argmax(transfer_values[i:j], 1) / tf.reduce_sum(tf.argmax(transfer_values[i:j], 1), 0))
cls_prob[i:j] = session.run(proba, feed_dict)
i = j
correct = (cls_true == cls_pred)
```

Hope this helps anybody out of trouble. ;-)
